### ISSUE TITLE
Support chains without uni v2

### ISSUE THREAD

**Issue by pandadefi** [Elnaril/uniswap-smart-path#1]
I had a look to add support for optimism, optimism doesn't have Uniswap v2, would you be ok to make the changes to disable Uniswap v2 if not available?

```
    42: (
        Token(Web3.to_checksum_address("0x7f5c764cbc14f9669b88837ca1490cca17c31607"), "USDC", 6),
        Token(Web3.to_checksum_address("0x94b008aA00579c1307B0EF2c499aD98a8ce58e58"), "USDT", 6),
        Token(Web3.to_checksum_address("0xDA10009cBd5D07dd0CeCc66161FC93D7c9000da1"), "DAI", 18),
        Token(Web3.to_checksum_address("0x4200000000000000000000000000000000000006"), "WETH", 18),
        Token(Web3.to_checksum_address("0x4200000000000000000000000000000000000042"), "OP", 18),
    )
```

**Comment by Elnaril** [comment]
That's a great idea! And thank you for your interest in this lib. :)
Actually, I planned to make it compatible with other blockchains but never had time. And not sure when I could do it ...

Maybe something like having the uniswap contract addresses by chain id in `_constants.py`, a bit like we have the pivot tokens. And having a couple of flags (one for v2 and one for v3) in the SmartPath class based on whether or not these addresses are configured.
And with this flag we could directly return an empty list in `_build_v2_path_list` and `_build_v3_path_list`.

Thanks for the addresses, BTW.

**Comment by pandadefi** [comment]
One great thing is uniswap v3 has the same address on most chains.

**Comment by Elnaril** [comment]
Hello @pandadefi 

I've just released the v0.2.0 which adds support for custom pools and blockchains (see `SmartPath.create_custom()`).
It should allow you to request only V3 pools on Optimism if you wish.

I'm closing this ticket as completed, but feel free to re-open it if you think it still does not work.
